tests/framework/validators/ValidatorTest.php

```diff
@@ -214,4 +214,11 @@ public function testAddError()
         $errors = $m->getErrors('attr_msg_val');
         $this->assertEquals('attr_msg_val::abc::param_value', $errors[0]);
     }
+
+    public function testGetActiveValidatorsForSafeAttributes()
+    {
+        $model = $this->getTestModel();
+        $validators = $model->getActiveValidators('safe_attr');
+        $this->assertInstanceOf('yii\validators\NumberValidator', array_shift($validators));
```

Comment:
``getActiveValidators`` has integer [keys.](https://github.com/yiisoft/yii2/blob/master/framework/base/Model.php#L428) It wouldn't help. One solution that I can come with is a loop of all ActiveValidators and if at least one of them is Instance of NumberValidator, then the test is passed. 
```php
public function testGetActiveValidatorsForSafeAttributes()
    {
        $model = $this->getTestModel();
        $validators = $model->getActiveValidators('safe_attr');
        $is_found = false;
        foreach ($validators as $v) {
            if ($v instanceof NumberValidator) {
                $is_found = true;
                break;
            }
        }
        $this->assertTrue($is_found);
    }
```
Maybe you can suggest a better solution?